Subject: Linux-RT 2.6.25.4-RT
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From khilman@mvista.com Wed May 28 15:18:40 2008
Date: Tue, 20 May 2008 15:45:49 -0700
From: Kevin Hilman <khilman@mvista.com>
To: linux-rt-users@vger.kernel.org
Cc: linux-omap@vger.kernel.org, Kevin Hilman <khilman@mvista.com>
Subject: [PATCH 4/5] RT: ARM: disable duplicate cmpxchg

FIXME: The ARM cmpxchg macro conflicts with the generic ones from
cmpxchg-local.

Signed-off-by: Kevin Hilman <khilman@mvista.com>
---
 include/asm-arm/atomic.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.25.4-rt4/include/asm-arm/atomic.h
===================================================================
--- linux-2.6.25.4-rt4.orig/include/asm-arm/atomic.h	2008-05-29 09:46:19.000000000 -0400
+++ linux-2.6.25.4-rt4/include/asm-arm/atomic.h	2008-05-29 09:47:33.000000000 -0400
@@ -146,7 +146,7 @@ static inline unsigned long __cmpxchg(vo
 		return wrong_size_cmpxchg(ptr);
 }
 
-#define cmpxchg(ptr,o,n)						\
+#define cmpxchg__disabled(ptr,o,n)						\
 ({									\
 	__typeof__(*(ptr)) _o_ = (o);					\
 	__typeof__(*(ptr)) _n_ = (n);					\
